src/WebJobs.Script/Workers/Profiles/SystemConditionProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal sealed class SystemConditionProvider : IWorkerProfileConditionProvider
```

Comment:
Any specific reason for naming it as `SystemConditionProvider`? May be naming it to `WorkerProfileConditionProvider` to keep it consistent with other implementations like `WorkerProfileManager`.